lib/api/route/messages.dart

```diff
@@ -193,6 +194,7 @@ Future<SendMessageResult> sendMessage(
     'content': RawParameter(content),
     if (queueId != null) 'queue_id': queueId,
     if (localId != null) 'local_id': localId,
+    if (supportsReadBySender) 'read_by_sender': true,
```

Comment:
Rather than hardcoding this to true within the API bindings, let's make it a parameter of this function, and then the caller passes true for it.

That way, the binding remains a thin layer exposing basically the same API as the server does itself, just in Dart. That in turn means the call sites in the application code (the code outside `lib/api/`) can be read more or less directly against the Zulip Server API. I think that's helpful for reducing the number of layers one needs to look through and making it easier to understand what the code is doing.

(This function does have one significant deviation from the underlying API, in the `destination` argument. What's happening there is basically that this part of the Zulip Server API isn't written in a way that's readily compatible with good clear static types, and so some rearrangement is needed in order to do a solid job of translating the API into Dart. But hard-coding this parameter is a different kind of matter, which is more about the specific needs and use cases we expect our application to have rather than about translating into Dart.)